src/index.js

```diff
@@ -117,22 +123,29 @@ class Node extends EventEmitter {
       }
     })
 
-    this.swarm.listen((err) => {
+    series([
+      (cb) => this.swarm.listen(cb),
+      (cb) => {
+        if (this.modules.discovery) {
+          each(this.modules.discovery, (d, cb) => {
+            d.start(cb)
+          }, cb)
+        }
+        cb()
+      },
+      (cb) => this.dht.start(cb)
+    ], (err) => {
       if (err) {
         return callback(err)
       }
+
       if (ws) {
-        this.swarm.transport.add(ws.tag || ws.constructor.name, ws)
+        this.swarm.transport.add(
+          ws.tag || ws.constructor.name, ws
+        )
       }
```

Comment:
transports (swarm) prep needs to happen first than discovery as discovery will trigger dials that use swarm.